### PR TITLE
Add is_complex

### PR DESCRIPTION
**Pull request by rikardn**
Adding `is_complex` test. This is a bit more complete than the previous test functions. It tests for singular points for most trigonometric functions, supports assumptions and `add`, `mul` and `pow`.

### CODE REVIEW

File: symengine/number.h [hunk 1]
```diff
@@ -147,6 +147,7 @@ tribool is_nonpositive(const Basic &b);
 tribool is_negative(const Basic &b);
 tribool is_nonnegative(const Basic &b);
 tribool is_real(const Basic &b, const Assumptions *assumptions = nullptr);
+tribool is_complex(const Basic &b, const Assumptions *assumptions = nullptr);
```
Comment (certik) [inline symengine/number.h:150]: Where is this method implemented? I didn't see it in this PR, but maybe I missed it. I was just curious.

**Review by certik — APPROVED** [review]
I think this looks good. Thanks for working on this.

File: symengine/number.h (hunk 1 above)
Comment (rikardn) [reply symengine/number.h:150]: It is in `test_visitors.cpp` together with the other "test"-functions related to numbers.

Comment (certik) [reply symengine/number.h:150]: Ah, I see it now. Thanks!

**Comment by rikardn** [comment]
Thanks  @ondrej for reviewing!

**Pull request merged**